File: Java/Tema3_Ejercicio10.java
/*
 * TEMA 3: ESTRUCTURAS CONDICIONALES - EJERCICIO 10
 * ==================================================
 * Escriba un programa que determine la calificación según la nota obtenida.
 */

import java.util.Scanner;

public class Tema3_Ejercicio10 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        System.out.print("Digite la nota (0-100): ");
        double nota = scanner.nextDouble();
        
        System.out.println("\n=== CALIFICACIÓN ===");
        System.out.println("Nota: " + nota);
        
        String calificacion;
        if (nota < 0 || nota > 100) {
            calificacion = "NOTA INVÁLIDA";
        } else if (nota >= 90) {
            calificacion = "A - Excelente";
        } else if (nota >= 80) {
            calificacion = "B - Muy Bueno";
        } else if (nota >= 70) {
            calificacion = "C - Bueno";
        } else if (nota >= 60) {
            calificacion = "D - Aceptable";
        } else {
            calificacion = "F - Insuficiente";
        }
        
        System.out.println("Calificación: " + calificacion);
        
        // Determinar si pasó
        if (nota >= 60 && nota <= 100) {
            System.out.println("Resultado: APROBADO");
        } else if (nota < 60) {
            System.out.println("Resultado: REPROBADO");
        }
        
        scanner.close();
    }
}

/*
 * SALIDA ESPERADA:
 * Digite la nota (0-100): 85
 * 
 * === CALIFICACIÓN ===
 * Nota: 85.0
 * Calificación: B - Muy Bueno
 * Resultado: APROBADO
 */

File: JS/Tema3_Ejercicio10.js
/*
 * TEMA 3: ESTRUCTURAS CONDICIONALES - EJERCICIO 10
 * =================================================
 * Asignar calificación (A-F) según puntuación.
 * 
 * EXPLICACIÓN:
 * - Rangos de puntuación
 * - Asignación de letras (A, B, C, D, F)
 * - Escala: A(90+), B(80+), C(70+), D(60+), F(<60)
 */

function asignarCalificacion(puntuacion) {
    if (puntuacion <0 || puntuacion > 100) {
        return "Puntuación inválida";
    }
    
    if (puntuacion >= 90) {
        return "A";
    } else if (puntuacion >= 80) {
        return "B";
    } else if (puntuacion >= 70) {
        return "C";
    } else if (puntuacion >= 60) {
        return "D";
    } else {
        return "F";
    }
}

console.log("=== ASIGNACIÓN DE CALIFICACIONES ===");
const puntuaciones = [95, 87, 76, 65, 45, 100, 59];

puntuaciones.forEach(puntaje => {
    const calificacion = asignarCalificacion(puntaje);
    console.log(`Puntuación: ${puntaje} → Calificación: ${calificacion}`);
});

/*
 * SALIDA ESPERADA:
 * === ASIGNACIÓN DE CALIFICACIONES ===
 * Puntuación: 95 → Calificación: A
 * Puntuación: 87 → Calificación: B
 * Puntuación: 76 → Calificación: C
 * Puntuación: 65 → Calificación: D
 * Puntuación: 45 → Calificación: F
 * Puntuación: 100 → Calificación: A
 * Puntuación: 59 → Calificación: F
 */
